xmtp_cursor_state/Cargo.toml

```diff
@@ -0,0 +1,10 @@
+[package]
+edition = "2021"
+name = "xmtp_cursor_state"
+rust-version = "1.70"
+version.workspace = true
+license.workspace = true
+
+[dependencies]
+hex = "0.4.3"
+xmtp_proto = { path = "../xmtp_proto", features = ["convert"] }
```

Comment:
for persistence, you can create migrations with the diesel crate (in `xmtp_db`) for sqlite table, then store a reference to the `XmtpDb` or just `ConnectionExt` generic trait on the store